src/DataStructures/Tensor/Expressions/TensorExpression.hpp

```diff
@@ -307,33 +307,27 @@ struct TensorExpression<Derived, DataType, Symm, IndexList, ArgsList<Args...>> {
   using args_list = ArgsList<Args...>;
 
   // @{
-  /// Cast down to the derived class. This is enabled by the
+  /// If Derived is a TensorExpression, it is casted down to the derived
+  /// class. This is enabled by the
   /// [CRTP](https://en.wikipedia.org/wiki/Curiously_recurring_template_pattern)
-  template <typename V = Derived,
-            Requires<not tt::is_a<Tensor, V>::value> = nullptr>
-  SPECTRE_ALWAYS_INLINE const Derived& operator~() const {
-    return static_cast<const Derived&>(*this);
-  }
-  // @}
-  // @{
-  /// If the Derived class is a Tensor return a const reference to a
-  /// TensorExpression.
   ///
-  /// Since Tensor is not derived from TensorExpression (because of
-  /// complications arising from the indices being part of the expression,
-  /// specifically Tensor may need to derive off of hundreds or thousands of
-  /// base classes, which is not feasible), return a reference to a
-  /// TensorExpression, which has a sufficient interface to evaluate the
-  /// expression.
+  /// Otherwise, it is a Tensor. Since Tensor is not derived from
+  /// TensorExpression (because of complications arising from the indices being
+  /// part of the expression, specifically Tensor may need to derive off of
+  /// hundreds or thousands of base classes, which is not feasible), return a
+  /// reference to a TensorExpression, which has a sufficient interface to
+  /// evaluate the expression.
+  ///
   /// \returns const TensorExpression<Derived, DataType, Symm, IndexList,
   /// ArgsList<Args...>>&
-  template <typename V = Derived,
-            Requires<tt::is_a<Tensor, V>::value> = nullptr>
-  SPECTRE_ALWAYS_INLINE auto operator~() const {
-    return static_cast<const TensorExpression<Derived, DataType, Symm,
-                                              IndexList, ArgsList<Args...>>&>(
-        *this);
+  SPECTRE_ALWAYS_INLINE const auto& operator~() const {
+    if constexpr (tt::is_a<Tensor, Derived>::value) {
```

Comment:
We have a shortcut for the `::value`, so you can just use `tt::is_a_v<Tensor, Derived>`

Reply:
done!